siliconcompiler/scheduler/taskscheduler.py

```
@@ -26,15 +36,15 @@ class TaskScheduler:
     dependency checking. It operates on a set of pending tasks defined by the
     main Scheduler and executes them in a loop until the flow is complete.
     """
-    __callbacks = {
+    __callbacks: Dict[str, Callable[["Project"], None]] = {
         "pre_run": lambda project: None,
         "pre_node": lambda project, step, index: None,
         "post_node": lambda project, step, index: None,
         "post_run": lambda project: None,
     }
```

Ruff (0.14.0)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)